### DISCUSSION TITLE
Is it possible to check if a partition has any rows through a c# script?

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#1199] [🙋♀️TE3 Q&A]
Hello, 
I'm creating partitions using a filter on a datetime column. 
Since I don't need the column for analysis, I'm not keeping it in the model (very large) 
But now I want to check if a partition has data
I did not see any property in the intellisense for partition, but hopefully there is some trick. I guess that the number of rows belongs to the metadata so it could be queried somehow 

Am I into something? 

Regards!

**Comment by bernatagulloesbrina** [comment]
Oh I see that theres a partition state, and this may be "NoData" or "Ready" etc
this may be enough for my use case 


I wonder how does SSMS get the rows in each partition though

**Reply by otykier** [reply]
NoData and Ready does not necessarily indicate if there are rows in the partition. Only whether or not it has been refreshed (a partition can have status Ready, even if it doesn’t contain any rows).

The TOM does not hold information about the row content. You’d need to execute a query, for example a DAX COUNTROWS to get statistics for the table, or a DMV (like VertiPaq Analyzer does) if you need statistics for individual partitions. In Tabular Editor, you can execute a query using the “ExecuteReader” or “ExecuteDax” C# script methods (the latter, contrary to its name, can also be used to execute MDX or DMV queries), https://docs.tabulareditor.com/te2/Useful-script-snippets.html#querying-analysis-services . Of course this technique only works while connected to an instance of AS.
